contracts/other/notional/Join1155.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/vault-interfaces/IJoin.sol";
+import "@yield-protocol/utils-v2/contracts/token/IERC20.sol";
+import "@yield-protocol/utils-v2/contracts/token/MinimalTransferHelper.sol";
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+import "@yield-protocol/utils-v2/contracts/math/WMul.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
+import "./ERC1155.sol"; // TODO: Move to yield-utils-v2
+// ERC1155TokenReceiver is in ERC1155.sol
+
+contract Join1155 is IJoin, ERC1155TokenReceiver, AccessControl() {
+    using WMul for uint256;
+    using CastU256U128 for uint256;
+
+    event FlashFeeFactorSet(uint256 indexed fee);
+
+    bytes32 constant internal FLASH_LOAN_RETURN = keccak256("ERC3156FlashBorrower.onFlashLoan");
+    uint256 constant FLASH_LOANS_DISABLED = type(uint256).max;
+
+    address public immutable override asset;
+    uint256 public immutable id;    // This ERC1155 Join only accepts one id from the ERC1155 token
+    uint256 public storedBalance;
+    uint256 public flashFeeFactor = FLASH_LOANS_DISABLED; // Fee on flash loans, as a percentage in fixed point with 18 decimals. Flash loans disabled by default.
+
+    constructor(address asset_, uint256 id_) {
+        asset = asset_;
+        id = id_;
+    }
+
+    /// @dev Advertising through ERC165 the available functions
+    function supportsInterface(bytes4 interfaceID) external view returns (bool) {
+        return  interfaceID == Join1155.supportsInterface.selector ||    // ERC-165 support (i.e. `bytes4(keccak256('supportsInterface(bytes4)'))`).
+                interfaceID == ERC1155TokenReceiver.onERC1155Received.selector ^ ERC1155TokenReceiver.onERC1155BatchReceived.selector;      // ERC-1155 `ERC1155TokenReceiver` support (i.e. `bytes4(keccak256("onERC1155Received(address,address,uint256,uint256,bytes)")) ^ bytes4(keccak256("onERC1155BatchReceived(address,address,uint256[],uint256[],bytes)"))`).
```

Comment:
this comment is a bit unwieldly can we move it to the next time or otherwise break it up?

Reply:
Fixed